### PR TITLE
chore : refactor code quality issues

### PR DESCRIPTION
**Pull request by powerexploit**
## Description
Hey 👋 , I'm member of the Developer Outreach team at DeepSource and ran DeepSource analysis on my fork of the repo. It found some interesting [code quality improvements](https://deepsource.io/gh/ankitdobhal/pymc3/issues) to consider.

This PR fixes a few of the issues detected for you to assess if it is right for you.
Happy to provide the tweaks separately otherwise :)

### What problem does this PR solve?

This PR improve some code quality issues like anti-pattern and bug risks. 

### What is changed and how it works?

- Remove `unused` imports
- Change equality `comparison` to identity comparison
- Added `.deepsource.toml` file for continuous analysis on bug risks/performance/code-quality issues on new changes.

### CODE REVIEW

File: pymc3/step_methods/sgmcmc.py [hunk 1]
```diff
@@ -139,7 +139,7 @@ def __init__(
         self.batch_size = batch_size
         self.total_size = total_size
         _value_error(
-            total_size != None or batch_size != None,
+            total_size is not None or batch_size != None,
```
Comment (ColCarroll) [inline pymc3/step_methods/sgmcmc.py:142]: Funny it misses the same problem a second time on this line.

**Review by ColCarroll — COMMENTED** [review]
These are good changes, but I haven't checked out the tool, and the tradeoffs to adding it to CI.

File: pymc3/step_methods/sgmcmc.py (hunk 1 above)
Comment (michaelosthege) [reply pymc3/step_methods/sgmcmc.py:142]: Shouldn't ths type of problem be caught by our existing pylint pre-commit?

Comment (brandonwillard) [reply pymc3/step_methods/sgmcmc.py:142]: If not, there are at least a couple established `pre-commit`-compatible tools that do exactly these things, so we would need to know why this tool is better.

Comment (powerexploit) [reply pymc3/step_methods/sgmcmc.py:142]: Their are following points that tells why **DeepSource** is better tool . 

- DeepSource analyzers detects in-depth issues than just surface level issues. 2000+ issue coverage.
- Ability to autofix issues and auto format code.
- DeepSource guarantees less than 5% false positives.
- Better integration and developer experience.

**Comment by twiecki** [comment]
The changes look OK but are very minor, not enough to warrant adding this to our CI.

**Pull request closed without merging**